plugins/catalog-backend/src/modules/core/PlaceholderProcessor.ts

```diff
@@ -127,6 +134,18 @@ export class PlaceholderProcessor implements CatalogProcessor {
           base,
         });
 
+      emit(
+        processingResult.refresh(
+          `url:${relativeUrl({
+            key: resolverKey,
+            value: resolverValue,
+            baseUrl: location.target,
+            read,
+            resolveUrl,
+          })}`,
+        ),
```

Comment:
I think this is gonna work for a lot of `resolver` implementations, but not all. I think it's worth an investigation into what it would mean to let the `resolver` return or emit refresh keys. Think for example an OpenAPI-aware resolver that follows `$ref` URLs that we then add to the set of refresh keys

Reply:
Agree, that's gonna be more future-proof for sure! Let me see how would it look.